cmd/neofs-adm/internal/modules/morph/initialize_deploy.go

```diff
@@ -551,20 +552,43 @@ func (c *initializeContext) getContractDeployData(ctrName string, keysParam []in
 			c.Contracts[netmapContract].Hash,
 			c.Contracts[containerContract].Hash)
 	case netmapContract:
-		configParam := []interface{}{
-			netmapEpochKey, viper.GetInt64(epochDurationInitFlag),
-			netmapMaxObjectSizeKey, viper.GetInt64(maxObjectSizeInitFlag),
-			netmapAuditFeeKey, viper.GetInt64(auditFeeInitFlag),
-			netmapContainerFeeKey, viper.GetInt64(containerFeeInitFlag),
-			netmapContainerAliasFeeKey, viper.GetInt64(containerAliasFeeInitFlag),
-			netmapEigenTrustIterationsKey, int64(defaultEigenTrustIterations),
-			netmapEigenTrustAlphaKey, defaultEigenTrustAlpha,
-			netmapBasicIncomeRateKey, viper.GetInt64(incomeRateInitFlag),
-			netmapInnerRingCandidateFeeKey, viper.GetInt64(candidateFeeInitFlag),
-			netmapWithdrawFeeKey, viper.GetInt64(withdrawFeeInitFlag),
-			netmapHomomorphicHashDisabledKey, viper.GetBool(homomorphicHashDisabledInitFlag),
-			netmapMaintenanceAllowedKey, viper.GetBool(maintenanceModeAllowedInitFlag),
+		configParam := []interface{}{}
+		for _, kf := range []struct {
+			key  string
+			flag string
+		}{
+			{netmapEpochKey, epochDurationInitFlag},
+			{netmapMaxObjectSizeKey, maxObjectSizeInitFlag},
+			{netmapAuditFeeKey, auditFeeInitFlag},
+			{netmapContainerFeeKey, containerFeeInitFlag},
+			{netmapContainerAliasFeeKey, containerAliasFeeInitFlag},
+			{netmapBasicIncomeRateKey, incomeRateInitFlag},
+			{netmapInnerRingCandidateFeeKey, candidateFeeInitFlag},
+			{netmapWithdrawFeeKey, withdrawFeeInitFlag},
+		} {
+			i64, err := unwrap.Int64(c.ReadOnlyInvoker.Call(ctrHash, "config", kf.key))
+			if err != nil {
+				i64 = viper.GetInt64(kf.flag)
```

Comment:
AFAIK netmap contract doesn't touch already set values. Maybe we can just check config setting via `viper.IsSet` and pass explicitly specified values only?

Reply:
Thanks, I think I got your point. 

Right now we must set all values explicitly since defaults are not defined. It's worth to think about them to move towards easier deployment